### PR TITLE
feat: trivy scan security reports in VHD builds

### PR DESCRIPTION
**Pull request by ganeshkumar5699**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Adds trivy security scanning reports to VHD builds. These will also be added to VHD release notes.

Every VHD build will generate two trivy artifacts per SKU, with list of components that have high and critical vulnerabilities, that have fixes available. These are the two files: 
- trivy-table.txt: human-readable file with list of SKUs for fast checking of affected components.
- trivy-report.json: machine readable file, that is very large, and contains additional information about affected components.

With the autonotes integration in this PR, these trivy reports will also be published and added to the released notes. 

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by ganeshkumar5699** [comment]
Based on: https://github.com/Azure/AgentBaker/pull/2930/files

**Comment by ganeshkumar5699** [comment]
Sample release notes with trivy reports: https://github.com/Azure/AgentBaker/pull/3085

File: vhdbuilder/release-notes/autonotes/main.go [hunk 1]
```diff
@@ -180,12 +198,58 @@ func getReleaseNotes(sku, path string, fl *flags, errc chan<- error, done chan<-
...
+		}
+		return
+	}
+
+	if err := os.Rename(trivyReportTableIn, trivyReportTableOut); err != nil {
+		errc <- fmt.Errorf("failed to rename file %s to %s, err: %s", trivyReportTableIn, trivyReportTableOut, err)
+		return
+	}
+
+	data, err = os.ReadFile(trivyReportTableOut)
```
Comment (alexeldeib) [inline vhdbuilder/release-notes/autonotes/main.go:245]: wait, what? isn't this the point of the os.Rename above?

Comment (alexeldeib) [reply vhdbuilder/release-notes/autonotes/main.go:245]: ohh misread. this is for latest.txt, nice addition

File: vhdbuilder/release-notes/autonotes/main.go [hunk 2]
```diff
@@ -180,12 +198,58 @@ func getReleaseNotes(sku, path string, fl *flags, errc chan<- error, done chan<-
 
 	data, err = os.ReadFile(imageListFileOut)
 	if err != nil {
-		errc <- fmt.Errorf("failed to read file %s for copying, err: %s", releaseNotesFileOut, err)
+		errc <- fmt.Errorf("failed to read file %s for copying, err: %s", imageListFileOut, err)
```
Comment (alexeldeib) [inline vhdbuilder/release-notes/autonotes/main.go:201]: nice

**Review by alexeldeib — APPROVED** [review]
think this lgtm, our release notes PRs about to balloon...72/4 * 6 = 108 files?

**Comment by ganeshkumar5699** [comment]
> think this lgtm, our release notes PRs about to balloon...72/4 * 6 = 108 files?

thanks. Yes, these trivy reports would increase it to 108 files 😊

**Review by juan-lee — APPROVED** [review]

**Pull request merged**